File: src/nodejs/IRISNative.js
const irisnative = require('intersystems-iris-native')

// Modify connection info based on environment
let connectionInfo = {
    host: '127.0.0.1', 
    port: 51773,
    ns: 'USER', 
    user: '_SYSTEM',
    pwd: 'SYS' 
};
// create database connection
const connection = irisnative.createConnection(connectionInfo);

//create IRIS instance
const dbnative = connection.createIris();

console.log('[1] Setting and getting a global');
// setting and getting a global
// ObjectScript equivalent: set ^nodejs("1") = 8888
dbnative.set(8888, 'nodejs', '1');

// ObjectScript equivalent: set globalValue = $get(^nodejs("1"))
let globalValue = dbnative.get('nodejs','1');
console.log('The value of nodejs is ' + globalValue);
console.log();

console.log('[2] Iterating over a global');
// modify global to iterate over
dbnative.set(7777, 'nodejs', '1');
dbnative.set(8888, 'nodejs', '2');
dbnative.set(9999, 'nodejs', '3');

let subscriptIter = dbnative.iterator('nodejs');  
console.log('walk forwards');
for ([key,value] of subscriptIter) {
    console.log('subscript='+ key +', value=' + value);
};
console.log();

console.log('Iterate backwards a different way');
let revIter = dbnative.iterator('nodejs').reversed();
let node = revIter.next();
while (!node.done) {
    console.log('subscript='+ node.value[0] +', value='+ node.value[1]);
    node = revIter.next();
}
console.log();

console.log('[3] Calling a class method');
// calling a class  method
// ObjectScript equivalent: set returnValue = ##class(%Library.Utility).Date(5)
let returnValue = dbnative.classMethodValue("%Library.Utility", "Date", 5);
console.log(returnValue);

// close connection
connection.close();

File: src/python/IRISNative.py
import irisnative

# Modify connection info based on your environment
ip = "127.0.0.1"
port = 51773
namespace = "USER"
username = "_SYSTEM"
password = "SYS"

# create database connection and IRIS instance
connection = irisnative.createConnection(ip,port,namespace,username,password)
dbnative = irisnative.createIris(connection)

print("[1. Setting and getting a global]")
# setting and getting a global
# ObjectScript equivalent: set ^python("1") = 8888
dbnative.set(8888, "python", "1")

# ObjectScript equivalent: set globalValue = $get(^python("1"))
globalValue = dbnative.get("python","1")

print("The value of python is ", globalValue)
print()

print("[2 Iterating over a global]")
# modify global to iterate over
# ObjectScript equivalent: set ^python("1") = 8888
# ObjectScript equivalent: set ^python("2") = 9999
dbnative.set(8888, "python", "1")
dbnative.set(9999, "python", "2")

Iter = dbnative.iterator("python")
print("walk forwards")
for subscript, value in Iter.items():
    print("subscript= {}, value={}".format(subscript, value))
print()

print("[3. Calling a class method]")
# calling a class  method
# ObjectScript equivalent: set returnValue = ##class(%Library.Utility).Date(5)
returnValue = dbnative.classMethodValue("%Library.Utility", "Date", 5)
print(returnValue)

# close connection
connection.close()